### DISCUSSION TITLE
Level object offsets

### DISCUSSION THREAD

**Discussion by city41** [city41/smaghetti#141] [Q&A]
Hey @nintyalex can you tell me more about [the level object offsets](https://github.com/city41/smaghetti/wiki/Level-Object-Offsets)?

What are they and how did you find them?

I'm currently taking a stab at getting 1-1 loaded into hextree. I am using the clue that TCRF gives that 1-1's first goomba is at `0x157812`. But that's well beyond all of these offsets. Does the game keep the level headers all together, and then maybe all level datas off in their own section?

**Comment by nintyalex** [comment]
Those of the offsets where each room's level data (usually terrain and decorations) starts. I found them by using the data found at the end of the "SMA4 Notes" file and using hex editor HxD and NES SMB3 editor SMB3 Workshop.

Each room's level data starts with its header, followed by the objects in said room.

The offset you used is actually that level's sprite data (mostly enemies).

**Reply by city41** [reply]
Ok nice. Today I made some good progress on level sprites. I am finding they mostly live a good ways away from the level offsets.

For the objects of a level, they appear to start at the level offsets you found + 15 bytes, where that 15 bytes looks to be a header. I noted this in the wiki page.

Hex-Tree now has a dropdown with a ton of hex values in it

![image](https://user-images.githubusercontent.com/141159/132070488-fda99e0b-25b8-4c28-b6f3-f0d6ca7ff3c7.png)

These hex values are likely the sprites to levels. I've already figured out 6 levels, and am currently working to figure out more. If anyone reading this knows any, please comment!

Some of the hex offsets in the dropdown are not levels, I am weeding those false positives out right now too.

Also in game levels don't look to have a "rooms" concept like e-reader levels do. Basically each "room" of an in game level is its own level as far as the game engine is concerned. This also looks to mean transports work very differently.
